backend/src/case_file/case_file.service.ts

```diff
@@ -904,5 +909,193 @@ export class CaseFileService {
+          actionCode: actionData.action_code,
+        });
+
+        equipmentDetailsMap.set(equipment.equipment_guid, equipmentDetail);
+      }
+    });
+    const equipmentDetails = Array.from(
+      equipmentDetailsMap.values()
+    ) as Equipment[];
+
+    return equipmentDetails;
+  }
+
+  async updateEquipment(
+    updateEquipmentInput: UpdateEquipmentInput
+  ): Promise<CaseFile> {
+
+    let caseFileOutput: CaseFile;
+    try {
+      await this.prisma.$transaction(async (db) => {
+      
+      });
+      } catch (exception) {
+        console.log("exception", exception);
+        throw new GraphQLError(
+          "Exception occurred. See server log for details",
+          {}
+        );
+      }
+
+    return caseFileOutput;
+  }
+
+  // create an equipment record - with actions to either set the equipment, or set and remove the equipment
+  async createEquipment(
+    createEquipmentInput: CreateEquipmentInput
+  ): Promise<CaseFile> {
+
+    let caseFileOutput: CaseFile;
+    try {
+      await this.prisma.$transaction(async (db) => {
+
+        let caseFile = await this.findOneByLeadId(
+          createEquipmentInput.leadIdentifier
+        );
+        let caseFileGuid;
+    
+        if (caseFile?.caseIdentifier) {
+          caseFileGuid = caseFile.caseIdentifier;
+        } else {
+          caseFileGuid = await this.createCase(createEquipmentInput);
+        }
+
+        const newEquipmentJSON = {
+          active_ind: true,
+          create_user_id: createEquipmentInput.createUserId,
+          create_utc_timestamp: new Date(),
+          update_user_id: createEquipmentInput.createUserId,
+          update_utc_timestamp: new Date(),
+          equipment_code:
+            createEquipmentInput.equipment[0].actionEquipmentTypeCode,
+          equipment_location_desc:
+            createEquipmentInput.equipment[0].address,
+          equipment_geometry_point:
+            createEquipmentInput.equipment[0].equipmentGeometryPoint,
+        }
+
+        this.logger.debug(`Creating equipment: ${JSON.stringify(newEquipmentJSON)}`);
+
+        // create the equipment record
+        const newEquipment = await db.equipment.create({
+          data: newEquipmentJSON
+        });
+
+        this.logger.debug(`New Equipment: ${JSON.stringify(newEquipment)}`);
+
+        // we can only create one equipment at a time, so just grab the first one.
+        const equipmentDetailsInstance =
+          createEquipmentInput.equipment[0];
+        const actions = equipmentDetailsInstance.actions;
+      
+        // get the actions associated with the creation of the equipment.  We may be setting an equipment, or setting and removing an equipment
+        for (const action of actions) {
+          this.logger.debug(`Actions: ${JSON.stringify}`)
+          let actionTypeActionXref =
+            await db.action_type_action_xref.findFirstOrThrow({
+              where: {
+                action_type_code: 'EQUIPMENT',
```

Comment:
you want to avoid using hardcoded values like this here, instead there's an `ACTION_TYPE_CODES` and `ACTION_TYPE` constant to use instead

Reply:
Done!